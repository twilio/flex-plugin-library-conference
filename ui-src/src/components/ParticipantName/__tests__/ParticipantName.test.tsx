import React from 'react';
import '@testing-library/jest-dom';
import * as Flex from '@twilio/flex-ui';
import ParticipantName from '../ParticipantName';
import { render } from '@testing-library/react';
import fetch from 'jest-fetch-mock';
import '@testing-library/jest-dom';

describe('ParticipantName', () => {
  beforeAll(() => {
    fetch.enableMocks();
  });
  beforeEach(() => {
    fetch.resetMocks();
    jest.spyOn(console, 'log').mockImplementation(() => {});
    jest.spyOn(console, 'warn').mockImplementation(() => {});
    jest.spyOn(console, 'error').mockImplementation(() => {});
  });
  it('renders Name view', () => {
    const wrapper = render(<ParticipantName listMode={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders NameList view', () => {
    const wrapper = render(<ParticipantName listMode={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders NameList view with participantType customer', () => {
    const mockTask = {
      attributes: {
        name: 'mockName',
      },
    };
    const mockParticipant = {
      participantType: 'customer',
    };
    const { getByText } = render(<ParticipantName listMode={true} task={mockTask} participant={mockParticipant} />);
    expect(getByText('mockName')).toBeInTheDocument();
  });

  it('renders NameList view with participantType unknown', () => {
    const mockTask = {
      attributes: {
        name: 'mockName',
      },
    };
    const mockParticipant = {
      participantType: 'unknown',
      callSid: 'CSxxxxxx',
    };
    fetch.mockRejectOnce('Mock Error string');
    const { getByText } = render(<ParticipantName listMode={true} task={mockTask} participant={mockParticipant} />);
    expect(getByText('Unknown')).toBeInTheDocument();
  });

  it('renders NameList view with participantType unknown and no callSid', () => {
    const mockTask = {
      attributes: {
        name: 'mockName',
      },
    };
    const mockParticipant = {
      participantType: 'unknown',
    };
    const { getByText } = render(<ParticipantName listMode={true} task={mockTask} participant={mockParticipant} />);
    expect(getByText('Unknown')).toBeInTheDocument();
  });
});
