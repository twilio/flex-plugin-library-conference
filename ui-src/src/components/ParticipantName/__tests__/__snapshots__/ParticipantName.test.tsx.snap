// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipantName renders Name view 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ParticipantCanvas-Name css-mml8bm"
      >
        Unknown
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ParticipantCanvas-Name css-mml8bm"
    >
      Unknown
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ParticipantName renders NameList view 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ParticipantCanvas-Name css-oe15en"
      >
        Unknown
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ParticipantCanvas-Name css-oe15en"
    >
      Unknown
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
